ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -157,14 +157,21 @@ import qualified Development.IDE.Types.Shake as Shake
 import           Development.IDE.GHC.CoreFile
 import           Data.Time.Clock.POSIX             (posixSecondsToUTCTime, utcTimeToPOSIXSeconds)
 import Control.Monad.IO.Unlift
+#if MIN_VERSION_ghc(9,3,0)
+import GHC.Unit.Module.Graph
+import GHC.Unit.Env
+#endif
+import Debug.Trace
```

Comment:
do we need an hlint rule to ban `Debug.Trace` from landing? @michaelpj

Reply:
:+1:

Reply:
https://github.com/haskell/haskell-language-server/pull/3000